src/server/pps/server/pipeline_controller.go

```diff
@@ -176,7 +174,9 @@ func (pc *pipelineController) tryFinish() {
 // 1. retrieves its full pipeline spec and RC into the 'Details' field
 // 2. makes whatever changes are needed to bring the RC in line with the (new) spec
 // 3. updates 'pipelineInfo', if needed, to reflect the action it just took
-func (pc *pipelineController) step(timestamp time.Time) (retErr error) {
+//
+// returns true if the pipeline is deleted, and the pipelineController can try to shutdown
+func (pc *pipelineController) step(timestamp time.Time) (isDelete bool, retErr error) {
 	log.Debugf("PPS master: processing event for %q", pc.pipeline)
 
 	// Handle tracing
```

Comment:
I'm not sure what was going on when I loaded this (the indented code was inside one of the GitHub code folds, so that might have had something to do with it), but the code seems totally fine now, so I'm marking this resolved.